jscomp/core/lam_compile_primitive.ml

```diff
@@ -78,6 +117,32 @@ let translate loc (cxt : Lam_compile_context.t) (prim : Lam_primitive.t)
           | _ -> E.runtime_call Js_runtime_modules.option "nullable_to_opt" args
           )
       | _ -> assert false)
+  | Pimport -> (
+      match args with
+      | [ e ] -> (
+          match e.expression_desc with
+          | _ -> (
+              match output_prefix with
+              | Some output_prefix -> (
+                  let output_dir = Filename.dirname output_prefix in
+
+                  let module_id, module_value =
+                    match module_of_expression e.expression_desc with
+                    | [ module_ ] -> module_
+                    | _ -> assert false
+                    (* TODO: graceful error message here *)
+                  in
+
+                  let path =
+                    Js_name_of_module_id.string_of_module_id module_id
+                      ~output_dir (get_module_system ())
+                  in
+
+                  match module_value with
+                  | Some value -> wrap_then (import_of_path path) value
+                  | None -> import_of_path path)
+              | None -> assert false))
```

Comment:
Is `None` possible? If so what should happen?

Reply:
Not possible, I will clean up